test/testVegaWrapper.js

```diff
@@ -417,3 +418,340 @@ describe('vegaWrapper', function() {
+            fail({ blah: 1 }, 'wikiraw');
+            pass('blah', { query: { pages: [{ revisions: [{ content: 'blah' }] }] } }, 'wikiraw');
+        });
+
+        it('wikidatasparql', function () {
+            fail({ error: 'blah' }, 'wikidatasparql');
+            fail({ blah: 1 }, 'wikidatasparql');
+            fail({ results: false }, 'wikidatasparql');
+            fail({ results: { bindings: false } }, 'wikidatasparql');
+            fail({ results: { bindings: 100 } }, 'wikidatasparql');
+            pass([], { results: { bindings: [] } }, 'wikidatasparql');
+            pass([{ int: 42, float: 42.5, geo: [42, 144.5] }, { uri: 'Q42' }], {
+                results: {
+                    bindings: [{
+                        int: {
+                            type: 'literal',
+                            'datatype': 'http://www.w3.org/2001/XMLSchema#int',
+                            value: '42'
+                        },
+                        float: {
+                            type: 'literal',
+                            'datatype': 'http://www.w3.org/2001/XMLSchema#float',
+                            value: '42.5'
+                        },
+                        geo: {
+                            type: 'literal',
+                            'datatype': 'http://www.opengis.net/ont/geosparql#wktLiteral',
+                            value: 'Point(42 144.5)'
+                        }
+                    }, {
+                        uri: {
+                            type: 'uri',
+                            value: 'http://www.wikidata.org/entity/Q42'
+                        }
+                    }]
+                }
+            }, 'wikidatasparql');
+        });
+
+        it('tabular', function () {
+            pass({
+                meta: [{
+                    description: 'desc',
+                    license_code: 'CC0-1.0+',
+                    license_text: 'abc',
+                    license_url: 'URL',
+                    sources: 'src'
+                }],
+                fields: [{ name: 'fld1' }],
+                data: [{ fld1: 42 }]
+            },
+                {
+                    jsondata: {
+                        description: 'desc',
+                        sources: 'src',
+                        license: { code: 'CC0-1.0+', text: 'abc', url: 'URL' },
+                        schema: { fields: [{ name: 'fld1' }] },
+                        data: [[42]]
+                    },
+                }, 'tabular');
+        });
+
+        it('map', function () {
+            pass({
+                meta: [{
+                    description: 'desc',
+                    license_code: 'CC0-1.0+',
+                    license_text: 'abc',
+                    license_url: 'URL',
+                    sources: 'src',
+                    longitude: 10,
+                    latitude: 20,
+                    zoom: 3,
+                }],
+                data: "map"
+            },
```

Comment:
if you delete a newline, (make it `}, {` ), the code will line up better. This is just a formatting recommendation.